src/libponyc/expr/operator.c

```diff
@@ -438,8 +438,13 @@ bool expr_assign(pass_opt_t* opt, ast_t* ast)
 
   if(!ok_safe)
   {
+    ast_t *left_lastchild = left;
+    while(ast_child(left_lastchild) != NULL)
+    {
+      left_lastchild = ast_child(left_lastchild);
+    }
     if(ast_id(left) == TK_FVARREF && ast_child(left) != NULL &&
-      ast_id(ast_child(left)) == TK_THIS)
+      ast_id(left_lastchild) == TK_THIS)
```

Comment:
This is definitely not correct. There's no reason to check if `ast_child(left)` isn't null here. You aren't using it.

Reply:
sorry that was from the previous code, ill remove the check